legate/core/solver.py

```diff
@@ -290,15 +295,17 @@ def cost(unknown):
             if isinstance(prev_part, NoPartition):
                 partition = prev_part
             else:
-                partition = store.compute_key_partition(restrictions)
+                if store._key_partition is not None:
+                    partition=store._key_partition
+                else:
+                    partition = store.compute_key_partition(restrictions)
```

Comment:
If the store already has a key partition (set by the fuser), no need to compute it 
There may be a more elegant way of doing this

Reply:
You should check if the key partition satisfies the restrictions before you reuse it. And I think this logic should be moved to `compute_key_partition`. Just so you know, I'm currently refactoring the code that manages legate stores, so don't make any micro optimization like this, as that will become obsolete in the near future.